src/components/frame/Extensions/pages/Extensions/ExtensionDetailsPage/partials/InstallButton.tsx

```diff
@@ -112,70 +72,32 @@ const InstallButton = ({
         toast.success(
           <Toast
             type="success"
-            title={{ id: 'extensionEnable.toast.title.success' }}
+            title={{ id: 'extensionInstallAndEnable.toast.title.success' }}
             description={{
-              id: 'extensionEnable.toast.description.success',
+              id: 'extensionInstallAndEnable.toast.description.success',
             }}
           />,
         );
 
         setActiveTab(ExtensionDetailsPageTabId.Settings);
-      } catch {
-        toast.error(
+      } else {
+        await waitForDbAfterExtensionAction({
+          method: ExtensionMethods.INSTALL,
+          refetchExtensionData,
+        });
+
+        setIsPolling(false);
```

Comment:
Looking at `setIsPolling(false)`, I think it's being called in all if-else blocks and in both the `catch ` and `try` blocks. Maybe we can just add this inside `finally`? But at the same time, maybe there's a reason why these have to be called at specific parts of the flow